dashboard/frontend/state/stores/runStore.js

```diff
@@ -0,0 +1,37 @@
+import { action, makeObservable, observable } from 'mobx';
+import Run from '../models/runModel';
+
+class RunStore {
+    @observable runs = [];
+
+    constructor(initialData = []) {
+        makeObservable(this);
+        this.runs = initialData;
+    }
+
+    @action newRun(run) {
+        this.runs.push(new Run(run));
+    }
+
+    @action setRuns(runs) {
+        this.runs = runs;
+    }
+
+    @action updateRunStatus(data, status) {
+        this.runs.filter((run) => {
```

Comment:
`filter` should not cause side-effects, use `forEach` instead.

Reply:
Better yet, use `find` and then update the status:
```suggestion
        const run = this.runs.find((run) => run.id === data.id);
        
        run && (run.status = status);
```